### ISSUE TITLE
Add Java and Scala dictionaries

### ISSUE THREAD

**Issue by arthurp** [streetsidesoftware/cspell-dicts#23]
Java and Scala are popular languages that are being used more and more with VS Code (in fact, Code is the only supported IDE for the Scala 3 prototype, Dotty). Simple dictionaries similar to what is provided for Python and other languages would be helpful to catch contractions and terms which are used in these languages.

**Comment by arthurp** [comment]
I have created raw word lists similar to what this repo contains for C++ and Python.

Java: https://gist.github.com/arthurp/91963552130d42a11cf7dc1ad1967c5b
Scala: https://gist.github.com/arthurp/20cd632ce0643a6ec3580276155bcc1c

To use them I applied the following local configuration:
```json
"enabledLanguageIds": [
    "scala",
    "java"
],
"dictionaryDefinitions": [
    { 
        "name": "scala", 
        "path": "./scala.txt",
        "description": "Scala standard library terms"
    },
    { 
        "name": "java", 
        "path": "./java.txt",
        "description": "Java standard library terms"
    }
],
"languageSettings": [
    {
        "languageId": "scala",
        "local": "*",
        "ignoreRegExpList": [
            "/import .*/"
        ],
         "dictionaries": ["scala"]
    },
    {
        "languageId": "java",
        "local": "*",
        "ignoreRegExpList": [
            "/import .*/"
        ],
        "dictionaries": ["java"]
    }
]
```

This is still rough, but the dictionaries themselves should be correct. The configuration probably needs some tweaking.

_You are free to use the above JSON and the content of the two gists with any license you like. I explicitly declaim copyright._

**Comment by Jason3S** [comment]
Thank you.

I'll see what I can do.

**Comment by Jason3S** [comment]
I added the dictionaries to the spell checker. The next time I publish the spell checker extension, I'll include Java and Scala by default.

**Comment by Jason3S** [comment]
This is what `cspell check` looks like:

![image](https://user-images.githubusercontent.com/3740137/41818772-72e20458-77b5-11e8-8d81-0f836ecf4e62.png)

**Comment by bentolor** [comment]
I currently stumbled over the issue, that Spell Checker raises it hands on the Vargs of `@SafeVargs` and I did not see any `java` dictionary available nor enabled.

1. What is the current state? From the comments it looks to me almost like most of the work is already done, no? What is missing to get `java` included into the list of available dictionaries and how can I help?
2. @arthurp It seems you wrote a script to extract public symbols from `rt.jar`. Which version? And any idea how we could get a complete list of  language keywords like the aforementioned  `@SafeVargs` ? I also stumble a little on values like `ABSTRACT`.

**Comment by arthurp** [comment]
>     2. @arthurp It seems you wrote a script to extract public symbols from `rt.jar`. Which version? And any idea how we could get a complete list of  language keywords like the aforementioned  `@SafeVargs` ? I also stumble a little on values like `ABSTRACT`.

The script may well work on any jar file. But I would definitely expect it to work on any version of rt.jar for sure.

`@SafeVargs` is not a keyword. There will be an interface in rt.jar named `SafeVargs`, so it should be picked up.

**Comment by bentolor** [comment]
> `@SafeVargs` is not a keyword. 

Of course you are right. Nevertheless the wordlist misses the keywords like `static` `int` `instanceof` etc.. Not to miss the literals `null`, `true`, `false`.

So probably just copy & paste them from https://docs.oracle.com/javase/tutorial/java/nutsandbolts/_keywords.html …

**Comment by bentolor** [comment]
I realized that @arthurp works directly on `rt.jar` and therefore is limited to Java 8 and older. It also missed the OpenJDK 12 symbols and also reserved words like `instanceof`. Therefore I created a small [Java application](https://github.com/bentolor/jdk9-module-enumerator) which tries to enumerate all public accessible symbols of JDK12+ via reflection. It was a little tricky at first due to the Jigsaw module management. The [classgraph project](https://github.com/classgraph/classgraph) turned out to be a valuable resource and help in my little quest… The result can be found in #60

Thanks again to @arthurp for going the first step! Improving is much more encouraging than starting…
